moto/elbv2/models.py

```diff
@@ -152,7 +152,17 @@ def deregister_terminated_instances(self, instance_ids: List[str]) -> None:
     def health_for(self, target: Dict[str, Any], ec2_backend: Any) -> FakeHealthStatus:
         t = self.targets.get(target["id"])
         if t is None:
-            raise InvalidTargetError()
+            port = self.port
+            if 'port' in target:
+                port = target["port"]
+            return FakeHealthStatus(
+                target["id"],
+                port,
+                self.healthcheck_port,
+                "unavailable",
```

Comment:
I am not sure about the status value, I assumed it for the desired `Reason=Target.NotRegistered`